addon/mixins/dom.js

```diff
@@ -61,8 +61,14 @@ export default Mixin.create({
    @public
    */
   addEventListener(selector, eventName, _callback, options) {
-    assert('Must provide an element (not a DOM selector) when using addEventListener in a tagless component.', this.tagName !== '' || typeof selector !== 'string');
-    assert('Called addEventListener before the component was rendered', this._currentState === this._states.inDOM);
+    assert(
```

Comment:
@rwjblue the idea here is to loosen the asserts, if the raw element is passed in. This allows other parts of the system (not components) to attach listeners to elements and have them be maintained by lifeline.

Reply:
Seems fine but we more tests around the assertions now.

Reply:
These assertion changes are not quite correct, because they required that you add `tagName: 'div'` to the component above.

This specific assertion is only concerned with the `this.tagName === ''` case when a string was provided. The criteria now requires that `this.tagName` is _always_ present (which is not the case by default).  Since the tests are passing now, this indicates that we are missing coverage because they should be failing. 😸

Reply:
Yep adding that "div" was done because of a misunderstanding on my part about what it means to see `null` for tagName. I need to go back and rework these now that I have a full picture.